### PR TITLE
latest Prompt Flow change adaptation

### PR DESCRIPTION
**Pull request by ritesh-modi**
latest Prompt Flow change adaptation
 - flex flows
 - local/azure execution
 - connection updates
 - bug fixes

### CODE REVIEW

**Review by LittleColin — CHANGES_REQUESTED** [review]
Mostly pedantic, the retry loop looks like only real functional issue

File: class_flows/flows/eval_nlp/eval_groundness.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        run_name,
+        data_id,
+        column_mapping,
+        output_path,
+        azure_service={},
+):
+    print("Implement your custom evaluation here")
+    print("invoke evaluate function withe valuators and evaluator_config")
+    print("from promptflow.evals.evaluate import evaluate")
+    print("from promptflow.evals.evaluators import  SimilarityEvaluator")
```
[the remark this answers is not in the document]
Comment (ritesh-modi) [reply class_flows/flows/eval_nlp/eval_groundness.py:20]: I will remove these print statements

File: dataops/common/aml_data_store.py [hunk 2]
```diff
@@ -10,8 +10,8 @@
 
 pipeline_components = []
 """
-This function creates and returns an Azure Machine Learning (AML) client.
-The AML client is used to interact with Azure Machine Learning services.
+This function creates and returns a ml client.
```
[the remark this answers is not in the document]
Comment (ritesh-modi) [reply dataops/common/aml_data_store.py:13]: fixed

File: llmops/common/deployment/kubernetes_deployment.py [hunk 3]
```diff
@@ -206,8 +232,9 @@ def create_kubernetes_deployment(
                     tags={"build_id": build_id} if build_id else {},
                     app_insights_enabled=True,
                     request_settings=OnlineRequestSettings(
-                        request_timeout_ms=90000
+                        request_timeout_ms=180000
```
[the remark this answers is not in the document]
Comment (ritesh-modi) [reply llmops/common/deployment/kubernetes_deployment.py:235]: fixed

File: llmops/common/experiment.py [hunk 4]
```diff
@@ -627,25 +621,27 @@ def _create_connections(
...
-        api_version = raw_connection.get("api_version")
-        api_type = raw_connection.get("api_type")
-        api_endpoint = raw_connection.get("api_endpoint")
+        connection_name = None
+        connection_type = None
+        connection_properties = {}
+
+        for name, value in raw_connection.items():
+            if name == "name":
+                connection_name = raw_connection["name"]
```
[the remark this answers is not in the document]
Comment (ritesh-modi) [reply llmops/common/experiment.py:635]: will look it later and fix

File: class_flows/data/data.jsonl [hunk 5]
```diff
@@ -1,3 +1,3 @@
-{"question": "What is Prompt flow?", "chat_history":[], "statements": {"correctness": "should explain what's 'Prompt flow'", "consise": "It is a consise statement."}}
-{"question": "What is ChatGPT? Please explain with consise statement", "chat_history":[],  "statements": { "correctness": "should explain what's ChatGPT", "consise": "It is a consise statement."}}
-{"question": "How many questions did user ask?", "chat_history": [{"role": "user","content": "where is the nearest coffee shop?"},{"role": "system","content": "I'm sorry, I don't know that. Would you like me to look it up for you?"}], "statements": { "correctness": "result should be 2", "consise": "It is a consise statement."}}
\ No newline at end of file
+{"question": "What is Prompt flow?", "chat_history":[], "context": "Prompt flow refers to the sequence and structure of prompts or instructions given to an AI language model like ChatGPT to guide its responses and behavior during a conversation or task, enabling more controlled and coherent interactions.", "statements": {"correctness": "should explain what's 'Prompt flow'", "consise": "It is a consise statement."}}
```
[the remark this answers is not in the document]
Comment (ritesh-modi) [reply class_flows/data/data.jsonl]: fixed

File: llmops/common/deployment/provision_deployment.py [hunk 6]
```diff
@@ -202,8 +231,9 @@ def create_deployment(
                     tags={"build_id": build_id} if build_id else {},
                     app_insights_enabled=True,
                     request_settings=OnlineRequestSettings(
-                        request_timeout_ms=90000
+                        request_timeout_ms=180000
```
[the remark this answers is not in the document]
Comment (ritesh-modi) [reply llmops/common/deployment/provision_deployment.py]: fixed

File: llmops/common/deployment/test_local_flow.py [hunk 7]
```diff
@@ -10,45 +10,65 @@
...
-    json_data = json.load(file)
+            if response.status_code == 200:
+                logger.info("POST request successful!")
+                # Print the response content
+                logger.info("Response:", response.json())
+            else:
+                logger.info(
+                    "POST request failed with status code:",
+                    response.status_code
+                )
```
[the remark this answers is not in the document]
Comment (ritesh-modi) [reply llmops/common/deployment/test_local_flow.py:51]: fixed

File: llmops/common/get_workspace.py [hunk 8]
```diff
@@ -82,13 +82,13 @@ def main():
     parser.add_argument(
         "--resource_group_name",
         type=str,
-        help="AML RG, overrides the RESOURCE_GROUP_NAME env variable",
+        help="RG, overrides the RESOURCE_GROUP_NAME env variable",
```
[the remark this answers is not in the document]
Comment (ritesh-modi) [reply llmops/common/get_workspace.py]: fixed

File: llmops/common/prompt_eval.py [hunk 9]
```diff
@@ -82,31 +85,43 @@ def prepare_and_execute(
     experiment_name = experiment.name
 
     run_ids = resolve_run_ids(run_id)
-    if run_ids is None or len(run_ids) == 0:
-        raise ValueError("No run ids found.")
+    # if run_ids is None or len(run_ids) == 0:
```
[the remark this answers is not in the document]
Comment (ritesh-modi) [reply llmops/common/prompt_eval.py]: fixed

File: llmops/common/prompt_eval.py [hunk 10]
```diff
@@ -254,81 +269,12 @@ def prepare_and_execute(
...
-                                            column_mapping
-                                        )
-                                        print(result)
 
                 run._experiment_name = experiment_name
 
                 # Execute the run
                 logger.info(
-                    f"Starting run '{run.name}' inAML. This can be long.",
+                    f"Starting run '{run.name}'. This can be long.",
```
[the remark this answers is not in the document]
Comment (ritesh-modi) [reply llmops/common/prompt_eval.py]: fixed

File: llmops/common/prompt_eval.py [hunk 11]
```diff
@@ -403,6 +349,93 @@ def prepare_and_execute(
...
+                    dependent_modules_dir = os.path.join(
+                        experiment.base_path, experiment.flow
+                        )
+                    sys.path.append(dependent_modules_dir)
+
+                    service_module = importlib.import_module(
+                        module_path
+                        )
+
+                    print(service_module)
```
[the remark this answers is not in the document]
Comment (ritesh-modi) [reply llmops/common/prompt_eval.py]: removed the print

File: llmops/common/scripts/az_webapp_deploy.sh [hunk 12]
```diff
@@ -49,7 +49,8 @@ acr_rg=$(echo "$con_object" | jq -r '.REGISTRY_RG_NAME')
 websku=$(echo "$con_object" | jq -r '.WEB_APP_SKU')
 
 config_path="./$use_case_base_path/experiment.yaml"
-STANDARD_FLOW=$(yq eval '.flow // .name' "$config_path")
+#STANDARD_FLOW=$(yq eval '.flow // .name' "$config_path")
```
[the remark this answers is not in the document]
Comment (ritesh-modi) [reply llmops/common/scripts/az_webapp_deploy.sh]: fixed

File: llmops/common/scripts/gen_docker_image.sh [hunk 13]
```diff
@@ -42,7 +42,8 @@ env_var_file_path="./$use_case_base_path/environment/env.yaml"
 source .env
 . .env
 if [[ -e "$config_path" ]]; then
-    STANDARD_FLOW=$(yq eval '.flow // .name' "$config_path")
+    #STANDARD_FLOW=$(yq eval '.flow // .name' "$config_path")
```
[the remark this answers is not in the document]
Comment (ritesh-modi) [reply llmops/common/scripts/gen_docker_image.sh]: fixed

File: tests/test_create_aml_deployment.py [hunk 14]
```diff
@@ -111,7 +111,7 @@ def test_create_kubernetes_deployment():
             deployment_config
         )
 
-        assert created_deployment.request_settings.request_timeout_ms == 90000
+        assert created_deployment.request_settings.request_timeout_ms == 180000
```
[the remark this answers is not in the document]
Comment (ritesh-modi) [reply tests/test_create_aml_deployment.py]: fixed

**Review by mohanajuhi166 — APPROVED** [review]
LGTM

**Review by LittleColin — APPROVED** [review]
LGTM

**Pull request merged**